### DISCUSSION TITLE
Syntax Highlighting for Type Checker Errors

### DISCUSSION THREAD

**Discussion by ilotoki0804** [microsoft/pyright#6260] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, type checker errors, along with other errors produced by Pyright, are not syntax highlighted. However, this leads to long or complex errors not being clearly visible.
![image](https://github.com/microsoft/pyright/assets/130233823/179be64d-f81f-41c0-962d-2dfb04bb0c89)

**Describe the solution you'd like**
I've discovered that syntax highlighting can markedly enhance the readability of errors. Thus, I suggest that syntax highlighting should be applied to type checker errors.

Current situation (Without syntax highlighting):
![image](https://github.com/microsoft/pyright/assets/130233823/42313037-6929-4e58-8f1d-3b987ff437d7)

Proposed improvement (With syntax highlighting):
![image](https://github.com/microsoft/pyright/assets/130233823/649f1789-5795-4e1f-8620-842cd5a21349)

**Comment by erictraut** [comment]
If you would like to suggest new language server features, please file those requests in the [pylance-release](https://github.com/microsoft/pylance-release) issue tracker. Pyright is focused on type checking features.

To my knowledge, there's no support in the language server protocol for providing syntax highlighting in diagnostic messages, so it's unlikely the pylance team would be able to do anything with this suggestion at this time.
